### PR TITLE
feat: basic keyboard shortcuts

### PR DESCRIPTION
**Pull request by hartmut-co-uk**
implements https://github.com/elk-zone/elk/issues/39

### TODOs
- [x] MVP
  - [x] add 'Keyboard Shortcuts' modal with instructions
      - [x] open via `?` -> `shift + ?`
      - [x] close modal via `shift + ?` / `esc`
  - [x] add first basic shortcut: `c` for 'Compose'
  - [x] add first two _sequence_ navigation shortcuts:
    - [x] `g..h` Open Home Timeline
    - [x] `g..n` Open Notifications
  - [x] technical details
    - [x] enable/disable based on _isAuthenticated_ status
    - [x] disable when focused on INPUT/TEXTBOX/contenteditable
  - [x] Toot _active_/_open_(?/_hover_?) specific actions
    - [x] Favourite `f`
    - [x] Boost `b`
  - [ ] ~navigate timeline prev/next status~ (moved to a future iteration due to complexity)
    - [ ] ~Next status `j`~
    - [ ] ~Prev status `k`~
  - [x] styling & design
    - [x] basic design 'Keyboard Shortcuts' modal
    - [x] responsive
- [x] define list of shortcuts to provide with this PR (limited number for a first iteration?)
- [x] use I18n (started off with _en-US_ for prototyping)
- [x] define next steps / create follow-up issue(s)   
        -> https://github.com/elk-zone/elk/issues/1855
        -> https://github.com/elk-zone/elk/issues/1856
        -> https://github.com/elk-zone/elk/issues/1857
        -> https://github.com/elk-zone/elk/issues/1858

### Options for improvement
- [ ] performance / ressource utilization
  - [ ] disable event listeners for mobile devices (without hardware keyboard)
- [ ] UI/UX
  - [ ] ??? for page navigation (e.g. `g .. h` / `g .. n`) show a notification message ~"Going to home" by @edimitchel 
- [ ] code / architecture
  - [ ] simplify / find better solution for `j`/`k` navigation.
         Current solution uses DOM & querySelectors, hard to maintain / comes with high risk of regressions while working on other features..
  - [ ] export composables for each keybinding, components self-register triggers, centralised tracking of listeners ([ref discord](https://discord.com/channels/1044887051155292200/1047780695105343530/1049603332404154388)) by @QiroNT

### CODE REVIEW

**Comment by jan-vandenberg** [comment]
The shortcut on Twitter for bookmark is 'b' (this will either add or remove a post from the bookmarks).
image

**Comment by MForster** [comment]
(I hope you don't mind drive-by commens. Feel free to ignore)

This is great!

One way it could be improved: Currently the scrolling with <kbd>j</kbd> / <kbd>k</kbd> is somewhat unpredictable. If the newly selected status still fits on the page only the focus is moved. If it doesn't fit, the page scrolls, and the focused status moves to the center of the page. This makes it hard to follow the focus with my eyes, because it's hard to predict where the next status will be drawn on the page after navigation.

IMHO there would be two alternative approaches that each would work better:
* Always scroll the newly focused status to the center, even if it isn't necessary
* Only scroll as far as needed that the newly focused status fits on the page "just so".

Both options would make it easier to follow the focus.

Just my 2¢...

**Comment by hartmut-co-uk** [comment]
I've added `z` key shortcut for 'zen mode', but I removed `j`/`k` for prev/next status navigation to do in a follow-up task.

#### Follow-up tasks with proposed mappings
- https://github.com/elk-zone/elk/issues/1855
- https://github.com/elk-zone/elk/issues/1856
- https://github.com/elk-zone/elk/issues/1857
- https://github.com/elk-zone/elk/issues/1858

Therefore with the basics added and a first setup to build on - I think this is ready to be reviewed.
I left TODOs in the code and would love to get feedback and/or suggestions.
'Favourite' and 'Boost' is not solved properly but working. Maybe it would be better to solve this properly _before_ merging this PR?

**Review by patak-dev — APPROVED** [review]
Thanks a lot for taking the time to work on this feature and split the PR into mergeable pieces @hartmut-co-uk 🧡
It works great for the current set of shortcuts. I think we should move forward and keep iterating with your plan.
